src/looprestoration.rs

```diff
@@ -23,3 +24,115 @@ pub union LooprestorationParams {
     pub filter: Align16<[[int16_t; 8]; 2]>,
     pub sgr: LooprestorationParams_sgr,
 }
+
+type pixel = libc::c_void;
+pub type const_left_pixel_row = *const libc::c_void; // *const [pixel; 4]
+
+pub type looprestorationfilter_fn = Option<
+    unsafe extern "C" fn(
+        *mut pixel,
+        ptrdiff_t,
+        const_left_pixel_row,
+        *const pixel,
+        libc::c_int,
+        libc::c_int,
+        *const LooprestorationParams,
+        LrEdgeFlags,
+        libc::c_int,
+    ) -> (),
+>;
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct Dav1dLoopRestorationDSPContext {
+    pub wiener: [looprestorationfilter_fn; 2],
+    pub sgr: [looprestorationfilter_fn; 3],
+}
+
+macro_rules! decl_looprestorationfilter_fn {
+    ( $( fn $name:ident, )* ) => {
+        extern "C" {
+            $(
+                // TODO(randomPoison): Temporarily pub until init fns are deduplicated.
+                pub(crate) fn $name(
+                    dst: *mut pixel,
+                    dst_stride: ptrdiff_t,
+                    left: const_left_pixel_row,
+                    lpf: *const pixel,
+                    w: libc::c_int,
+                    h: libc::c_int,
+                    params: *const LooprestorationParams,
+                    edges: LrEdgeFlags,
+                    bitdepth_max: libc::c_int,
+                );
+            )*
+        }
+    };
+}
+
+#[cfg(all(
+    feature = "bitdepth_8",
+    feature = "asm",
+    any(target_arch = "x86", target_arch = "x86_64"),
+))]
+decl_looprestorationfilter_fn! {
```

Comment:
If you keep it this way, could you make it plural instead: `decl_looprestorationfilter_fns!`.

Reply:
Done!